main_test.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"testing"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_runScriptBash(t *testing.T) {
+	t.Log("Successful execution")
+	{
+		require.NoError(t, fileutil.WriteStringToFile("_tmp/test.sh", "echo 'This is a Bash script'"))
+
+		exitCode, err := runScript("bash", "_tmp/test.sh")
+		require.Equal(t, 0, exitCode)
+		require.Equal(t, nil, err)
+	}
+
+	t.Log("Exit with code 222")
+	{
+		require.NoError(t, fileutil.WriteStringToFile("_tmp/test.sh", "exit 222"))
+
+		exitCode, err := runScript("bash", "_tmp/test.sh")
+		require.Equal(t, 222, exitCode)
+		require.Error(t, err)
+	}
+}
+
+func Test_runScriptRuby(t *testing.T) {
+	require.NoError(t, fileutil.WriteStringToFile("_tmp/test.rb", "puts 'This is a Ruby script'"))
+
+	exitCode, err := runScript("ruby", "_tmp/test.rb")
+	require.Equal(t, 0, exitCode)
+	require.Equal(t, nil, err)
+}
+
+func Test_runScriptGo(t *testing.T) {
+	goScript := heredoc.Doc(`
```

Comment:
why do we need heredoc.Doc?

this could be simply:

```
	goScript := `package main

import (
	"fmt"
)

func main() {
	fmt.Println("This is a Go script")
}
`
```